modules/phase_field/src/materials/RegularSolutionFreeEnergy.C

```diff
@@ -33,14 +33,13 @@ RegularSolutionFreeEnergy::RegularSolutionFreeEnergy(const InputParameters & par
     _omega(getParam<Real>("omega")),
     _kB(getParam<Real>("kB"))
 {
-  EBFunction free_energy;
+  EBTerm free_energy;
   // Definition of the free energy for the expression builder
-  free_energy(_c) =
-      _omega * _c * (1.0 - _c) + _kB * _T * (_c * log(_c) + (1.0 - _c) * log(1.0 - _c));
+  free_energy = _omega * _c * (1.0 - _c) + _kB * _T * (_c * log(_c) + (1.0 - _c) * log(1.0 - _c));
 
   // Use Taylor expanded logarithm?
-  if (isParamValid("log_tol"))
-    free_energy.substitute(EBLogPlogSubstitution(getParam<Real>("log_tol")));
+  // if (isParamValid("log_tol"))
+  //  free_energy.substitute(EBLogPlogSubstitution(getParam<Real>("log_tol")));
```

Comment:
we need this!

Reply:
Yes, I am planning on making this an automatic substitution in the PrepEBSimpleTrees. Should I make it automatic like this, or instead have it so that the user decides when to call it?

Reply:
Also, when taking a derivative we get a 1/x, should I create a new type of operator that has some tolerance for this as well?

Reply:
The derivative of `plog` is not `1/x`! Substitution needs to be performed before taking a derivative.